lib/ansible/module_utils/network/skydive/skydive.py

```diff
@@ -0,0 +1,171 @@
+#      notice, this list of conditions and the following disclaimer.
+#    * Redistributions in binary form must reproduce the above copyright notice,
+#      this list of conditions and the following disclaimer in the documentation
+#      and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+# IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+# USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+
+from ansible.module_utils.six import iteritems
+from ansible.module_utils._text import to_text
+from ansible.module_utils.basic import env_fallback
+
+try:
+    from skydive.graph import Node, Edge
+    from skydive.rest.client import RESTClient
+    from skydive.websocket.client import WSClient
+    from skydive.websocket.client import WSClientDebugProtocol
+    from skydive.websocket.client import WSMessage
+    from skydive.websocket.client import NodeAddedMsgType, EdgeAddedMsgType
+    HAS_SKYDIVE_CLIENT = True
+except ImportError:
+    HAS_SKYDIVE_CLIENT = False
+
+# defining skydive constants
+SKYDIVE_LOOKUP_QUERY = 'G.V().Has'
+
+SKYDIVE_PROVIDER_SPEC = {
+    'host': dict(fallback=(env_fallback, ['SKYDIVE_HOST'])),
+    'username': dict(fallback=(env_fallback, ['SKYDIVE_USERNAME'])),
+    'password': dict(fallback=(env_fallback, ['SKYDIVE_PASSWORD']), no_log=True),
+    'ssl_verify': dict(type='bool', default=False, fallback=(env_fallback, ['SKYDIVE_SSL_VERIFY']))
+}
+
+
+class skydive_base(object):
+    ''' Base class for implementing Skydive Rest API '''
+    provider_spec = {'provider': dict(type='dict', options=SKYDIVE_PROVIDER_SPEC)}
+
+    def __init__(self, host="localhost:8082", username="admin", password="password"):
+        try:
+            self.restclient_object = RESTClient(host, username=username, password=password)
+            self.websocketclient_object = WSClient(host, username=username, password=password)
+        except TypeError as err:
+            raise err
+
+
+class skydive_lookup(skydive_base):
+
+    def __init__(self, provider):
+        host = provider['host']
+        username = provider['username']
+        password = provider['password']
+        super(skydive_lookup, self).__init__(host, username, password)
+        self.query_str = ""
+
+    def lookup_query(self, filter_data):
```

Comment:
Limited to nodes and limited to a single Has filter. I would not constrain the gremlin query that can be sent. If you want to
provide a simpler lookup for people who do not want to understand gremlin, provide it as an additional query. Example of things that are useful in practice : Find the interface node (type "interface") named "eth0' that is a child of a host (type "host") that is named "my_machine". When you glue nodes created by ansible and nodes found by skydive, you need at least this expressive power in your lookup queries.